web-components/src/components/tooltip/Tooltip.ts

```diff
@@ -69,59 +69,16 @@ export class Tooltip extends FocusMixin(LitElement) {
     this.notifyTooltipDestroy();
   }
 
-  private openTooltip() {
+  private notifyVirtualTooltip(eventName: string, detailArgs?: Pick<TooltipEvent, "slotContent">) {
     this.dispatchEvent(
-      new CustomEvent<TooltipEvent>("tooltip-create", {
+      new CustomEvent<TooltipEvent>(eventName, {
         bubbles: true,
         composed: true,
         detail: {
           placement: this.placement,
           reference: this.reference,
           popper: this.popper,
-          ...(!this.message && { slotContent: this.slotContent })
-        }
-      })
-    );
-  }
-
-  private closeTooltip() {
-    this.dispatchEvent(
-      new CustomEvent<TooltipEvent>("tooltip-destroy", {
-        bubbles: true,
-        composed: true,
-        detail: {
-          placement: this.placement,
-          reference: this.reference,
-          popper: this.popper
-        }
-      })
-    );
-  }
-
-  private changeMessage() {
-    this.dispatchEvent(
-      new CustomEvent<TooltipEvent>("tooltip-message", {
-        bubbles: true,
-        composed: true,
-        detail: {
-          placement: this.placement,
-          reference: this.reference,
-          popper: this.popper
-        }
-      })
-    );
-  }
-
-  private changeSlotContent(slotContent: Element[]) {
```

Comment:
will removing this method make it impossible to change slotted content dynamically? do we have UT for it? do we have an example in Sandbox for it?

Reply:
@yananym please look at method:

> notifyVirtualTooltip
 
I just remove duplicated peaces of code. It handles all cases you mention.